beer-garden/beer-garden
# #1302 - Role Syncing

Closes #1302 

This PR adds functionality for role syncing across gardens.  The role syncing now happens as part of the user syncing, with the roles being imported on the remote garden first, followed by the users.  This ensures that user permissions are truly consistent across the gardens, rather than leaving open the possibility that users would have the same assigned roles, but that those role assignments might mean different things on different gardens.

## Test Instructions
The easiest way to test is to use the user sync option in the UI.

- Stand up a local and remote garden. Make sure auth is enabled.
- Make some changes to the remote garden so that the users and roles are not in sync with the local garden.
- Use the user sync functionality in the UI to sync the users and roles
- Check the remote garden and ensure that the users and roles are now in sync with the local.
